docker/daemon/client.go

```diff
@@ -0,0 +1,65 @@
+package daemon
+
+import (
+	"github.com/containers/image/types"
+	"github.com/docker/docker/api"
+	dockerclient "github.com/docker/docker/client"
+	"github.com/docker/go-connections/tlsconfig"
+	"net/http"
+	"path/filepath"
+)
+
+// NewDockerClient initializes a new API client based on the passed SystemContext.
+func newDockerClient(systemCtx *types.SystemContext) (*dockerclient.Client, error) {
+	if systemCtx == nil {
+		return dockerclient.NewClient(dockerclient.DefaultDockerHost, api.DefaultVersion, nil, nil)
```

Comment:
Does using `api.DefaultVersion` here, instead of hard-coding "1.22", work? Wouldn’t that mean that random rebases would break compatibility with earlier servers?

Cc: @runcom

Reply:
> Wouldn’t that mean that random rebases would break compatibility with earlier servers

Not sure what you mean with random rebases? I find "1.22" random tbh. I think the default should be the default as per used library. If you need a different version, you should set it explicitly.

Reply:
A ”random rebase” = the `docker/docker` package is re-vendored to any newer version, for whatever reason (new feature needed elsewhere, a security fix, whatever).

The API version is not a magic string that one should cargo-cult from `api.DefaultVersion`; it is used to determine compatibility, and always using the latest one seems unwarranted.

To be fair, it’s entirely possible that `1.22` is an incorrect value (it’s likely enough a frozen value of `api.DefaultVersion` as of one time) and the code could in fact work with an earlier version; but that doesn’t make always using the latest one any more correct.

Reply:
> To be fair, it’s entirely possible that 1.22 is an incorrect value (it’s likely enough a frozen value of api.DefaultVersion as of one time)

Right, that's what I think as well. It seems neither 1.22 nor api.DefaultVersion might be ideal, but given the choice, I'd go for the latter.

That said, I could introduce a constant within this package and we can decide whether we want to set it to "1.22" or _api.DefaultVersion_. 

and the code could in fact work with an earlier version; but that doesn’t make always using the latest one any more correct.

Reply:
> and the code could in fact work with an earlier version; but that doesn’t make always using the latest one any more correct.

It _does_ make it less useful, though, because using an earlier API version allows the client to work with more versions of the server. The lower the version specified by the client, the more server versions work; that also makes `api.DefaultVersion`, which is always the highest version available, the worst possible choice if I understand the situation correctly.

(Unless different versions of the API could be equally compatible but have different semantics, which is I’ve asked @runcom to confirm.)

Reply:
hmm, not sure, but I guess we can just keep the 1.22. I would love to hear @runcom's opinion as well